### PR TITLE
DATAUP-310 state - add a set of states, fix startup

### PR DESCRIPTION
**Pull request by briehl**
# Description of PR purpose/changes

This adds and annotates a set of states the cell will likely encounter, roughly matching the app cell. It modifies the cell metadata somewhat, so making a new bulk import cell (again, sorry) will be necessary. Now this should serialize the state, and also save the last visited tab.

It also fixes an issue where loading a page wouldn't start up the right tab at the right time.

# Jira Ticket / Issue #
e.g. https://kbase-jira.atlassian.net/browse/DATAUP-X
- [x] Added the Jira Ticket to the title of the PR e.g. (DATAUP-69 Adds a PR template)

# Testing Instructions
* Details for how to test the PR: 
- Make a new bulk import cell. Toggle the tab, save the Narrative, and reload.

- [x] Tests pass in travis and locally 
- [x] Changes available by spinning up a local narrative and navigating to _X_ to see _Y_

# Dev Checklist:

- [x] My code follows the guidelines at https://sites.google.com/truss.works/kbasetruss/development
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: nbextensions/bulkImportCell/bulkImportCellStates.js [hunk 1]
```diff
@@ -40,8 +61,346 @@ define([], () => {
...
+                            visible: true
+                        },
+                        results: {
+                            enabled: true,
+                            visible: true
+                        },
+                        error: {
+                            enabled: false,
+                            visible: false
+                        }
```
Comment (ialarmedalien) [inline nbextensions/bulkImportCell/bulkImportCellStates.js:244]: This configuration pops up numerous times -- is it worth saving it as a variable (`defaultRunConfig` or something) or using a function to generate it? There are also several cases above where the config is this, with one value changed to `false`; could it be created by copying the defaultRunConfig and altering whichever value is relevant?

File: test/unit/spec/nbextensions/bulkImportCell/bulkImportCell-spec.js [hunk 2]
```diff
@@ -53,6 +53,10 @@ define([
             }
             expect(cell.metadata.kbase.type).toBe('app-bulk-import');
             expect(cell.metadata.kbase.bulkImportCell).toBeDefined();
+            expect(cell.metadata.kbase.bulkImportCell.state).toEqual({
+                state: 'editingIncomplete',
+                selectedTab: 'configure'
+            });
```
Comment (ialarmedalien) [inline test/unit/spec/nbextensions/bulkImportCell/bulkImportCell-spec.js:59]: will there be additional tests where you move through different cell states?

Comment (briehl) [reply test/unit/spec/nbextensions/bulkImportCell/bulkImportCell-spec.js:59]: Yes, once there's a way to toggle through the different states. This is really just to get a few different states in place, and fix a startup bug.

File: nbextensions/bulkImportCell/bulkImportCellStates.js (hunk 1 above)
Comment (briehl) [reply nbextensions/bulkImportCell/bulkImportCellStates.js:244]: Sure, I'll add that.

Comment (briehl) [reply nbextensions/bulkImportCell/bulkImportCellStates.js]: @ialarmedalien ok, I turned that from a giant declared JSON structure to something assembled from functions.

**Review by ialarmedalien — APPROVED** [review]
Looks great! Merge away (once the conflicts are resolved).

**Pull request merged**